code/Model/Submission/SubmittedFileField.php

```diff
@@ -41,14 +41,24 @@ public function getFormattedValue()
         $name = $this->getFileName();
         $link = $this->getLink();
         $title = _t(__CLASS__ . '.DOWNLOADFILE', 'Download File');
+        $message = _t(__CLASS__ . '.INSUFFICIENTRIGHTS', 'You don\'t have the right permissions to download this file');
+        $file = $this->getUploadedFileFromDraft();
 
         if ($link) {
-            return DBField::create_field('HTMLText', sprintf(
-                '%s - <a href="%s" target="_blank">%s</a>',
-                $name,
-                $link,
-                $title
-            ));
+            if ($file->canView()) {
+                return DBField::create_field('HTMLText', sprintf(
+                    '%s - <a href="%s" target="_blank">%s</a>',
+                    htmlspecialchars($name),
+                    htmlspecialchars($link),
+                    htmlspecialchars($title)
+                ));
+            } else {
+                return DBField::create_field('HTMLText', sprintf(
+                    '<span class="icon font-icon-lock">%s - <i>%s</i></span>',
```

Comment:
I think that's the expected way to include an icon. We apply it to the `<i>` tag because it doesn't have any semantic meaning.
```suggestion
                    '<i class="icon font-icon-lock"></i> %s - <em>%s</em>',
```

Reply:
DONE